src/background/redux/epics/portEpic.ts

```diff
@@ -41,11 +43,37 @@ const portDisconnectEpic = (
   {router}: EpicDependencies
 ): Observable<Action<any>> =>
   action$.pipe(
-    ofType<Action<Runtime.Port>>(actions.PORT_DISCONNECT),
+    ofType<Action<Runtime.Port>>(actions.PORT_DISCONNECT), // TODO:
     switchMap(({payload}) => {
       router.remove(payload);
       return EMPTY;
     })
   );
 
-export default combineEpics(portConnectEpic, portDisconnectEpic);
+const enabledPortAddEpic = (
+  action$: ActionsObservable<Action<any>>,
+  state$: StateObservable<BackgroundState>,
+  {router}: EpicDependencies
+): Observable<Action<any>> =>
+  action$.pipe(
+    ofType<Action<string>>(PopupMsgTypes.ENABLED_PORT_ADD),
+    switchMap(({payload}) => {
+      router.enabledPortNames.push(payload)
+      return EMPTY;
+    })
+  );
+
+const enabledPortRemoveEpic = (
+  action$: ActionsObservable<Action<any>>,
+  state$: StateObservable<BackgroundState>,
+  {router}: EpicDependencies
+): Observable<Action<any>> =>
+  action$.pipe(
+    ofType<Action<string>>(actions.ENABLED_PORT_REMOVE),
+    switchMap(({payload}) => {
+      router.enabledPortNames = router.enabledPortNames.filter(name => name !== payload)
```

Comment:
better implement addEnabledPort and removeEnabledPort in router, rather than allowing outer code to operate directly on it

Reply:
Okay👌